Userland/Libraries/LibWeb/HTML/Worker.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2022, Ben Abraham
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/String.h>
+#include <LibWeb/Bindings/WorkerWrapper.h>
+#include <LibWeb/HTML/Worker.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/workers.html#dedicated-workers-and-the-worker-interface
+
+Worker::Worker(String& script_url, WorkerOptions const& options)
+    : Worker(script_url, options, nullptr)
+{
+}
+
+Worker::Worker(String& script_url, WorkerOptions const& options, DOM::Document* document)
+    : m_script_url(script_url)
+    , m_options(options)
+    , m_document(document)
+{
+    dbgln("Worker: Create={}", script_url);
+
```

Comment:
To construct objects like this "safely" or at least be able to throw the errors we usually do it in a static `create` method. See for example `ClassicScript::create(...)`

This will allow you to actually return/throw a DOMException and call any method with `TRY` if needed.

Reply:
I believe that I transferred this to the ::create properly, please review though. Still a little shaky on the means of raising exceptions.

Reply:
Looks good to me!